sdk/openai/Azure.AI.OpenAI/README.md

```diff
@@ -301,26 +301,70 @@ calls, and the tool messages that resolved each of those tools -- when making a
-    var toolCallHistoryMessage = new ChatRequestAssistantMessage(responseChoice.Message.Content);
-    foreach (ChatCompletionsToolCall requestedToolCall in responseChoice.Message.ToolCalls)
+    // Now make a new request with all the messages thus far, including the original
+}
+```
+
+When using tool calls with streaming responses, accumulate tool call details much like you'd accumulate the other
+portions of streamed choices, in this case using the accumulated `StreamingToolCallUpdate` data to instantiate new
+tool call messages for assistant message history. Note that the model will ignore `ChoiceCount` when providing tools
+and that all streamed responses should map to a single, common choice index in the range of `[0..(ChoiceCount - 1)]`.
+
+```C# Snippet:ChatTools:StreamingChatTools
+Dictionary<int, string> toolCallIdsByIndex = new();
+Dictionary<int, string> functionNamesByIndex = new();
+Dictionary<int, StringBuilder> functionArgmentBuildersByIndex = new();
+StringBuilder contentBuilder = new();
+
+await foreach (StreamingChatCompletionsUpdate chatUpdate
+    in await client.GetChatCompletionsStreamingAsync(chatCompletionsOptions))
+{
+    if (chatUpdate.ToolCallUpdate is StreamingFunctionToolCallUpdate functionToolCallUpdate)
     {
-        toolCallHistoryMessage.ToolCalls.Add(requestedToolCall);
+        if (functionToolCallUpdate.Id != null)
+        {
+            toolCallIdsByIndex[functionToolCallUpdate.ToolCallIndex] = functionToolCallUpdate.Id;
+        }
+        if (functionToolCallUpdate.Name != null)
+        {
+            functionNamesByIndex[functionToolCallUpdate.ToolCallIndex] = functionToolCallUpdate.Name;
+        }
+        if (functionToolCallUpdate.ArgumentsUpdate != null)
+        {
+            StringBuilder argumentsBuilder
+                = functionArgmentBuildersByIndex.TryGetValue(
+                    functionToolCallUpdate.ToolCallIndex,
+                    out StringBuilder existingBuilder) ? existingBuilder : new StringBuilder();
+            argumentsBuilder.Append(functionToolCallUpdate.ArgumentsUpdate);
+        }
     }
-
-    // Now make a new request using all the messages, including the original
-    chatCompletionsOptions.Messages.Add(toolCallHistoryMessage);
-    foreach (ChatRequestToolMessage resolutionMessage in toolCallResolutionMessages)
+    if (chatUpdate.ContentUpdate != null)
     {
-        chatCompletionsOptions.Messages.Add(resolutionMessage);
+        contentBuilder.Append(chatUpdate.ContentUpdate);
     }
 }
+
+ChatRequestAssistantMessage assistantHistoryMessage = new(contentBuilder.ToString());
+foreach (KeyValuePair<int, string> indexIdPair in toolCallIdsByIndex)
+{
+    assistantHistoryMessage.ToolCalls.Add(new ChatCompletionsFunctionToolCall(
+        id: indexIdPair.Value,
+        functionNamesByIndex[indexIdPair.Key],
+        functionArgmentBuildersByIndex[indexIdPair.Key].ToString()));
+}
+chatCompletionsOptions.Messages.Add(assistantHistoryMessage);
```

Comment:
Does this work without https://github.com/Azure/azure-sdk-for-net/pull/40704/files#r1429009644 ?